src/metabase/pulse/render/datetime.clj

```diff
@@ -50,15 +51,37 @@
         ts (u.date/parse "2022-01-01-00:00:00")]
     (u.date/format time-style (t/plus ts (t/hours n)))))
 
+(defn- viz-settings-for-col
+  "Get the column-settings map for the given column from the viz-settings."
+  [col viz-settings]
+  (let [[_ field-id]    (:field_ref col)
+        all-cols-settings (-> viz-settings
+                              ::mb.viz/column-settings
+                              ;; update the keys so that they will have only the :field-id or :column-name
+                              ;; and not have any metadata. Since we don't know the metadata, we can never
+                              ;; match a key with metadata, even if we do have the correct name or id
+                              (update-keys #(select-keys % [::mb.viz/field-id ::mb.viz/column-name])))]
+    (or (all-cols-settings {::mb.viz/field-id field-id})
+        (all-cols-settings {::mb.viz/column-name field-id}))))
```

Comment:
Reminiscent of `body/column-name` but maybe I won't lose sleep over any duplication